test/unit/engine/utils/create-patch.test.ts

```diff
@@ -9,15 +9,20 @@ describe('createPatch', () => {
     const targetEntry = targetEntriesFixture.items[0] as unknown as Entry<any>
     const patch = createPatch({ targetEntry, sourceEntry })
     expect(patch).to.deep.equal([
-      { op: 'replace', path: '/fields/slug/en-US', value: 'home-page' },
       { op: 'replace', path: '/fields/title/en-US', value: 'Home page' },
+      { op: 'replace', path: '/fields/slug/en-US', value: 'home-page' },
     ])
   })
   it('creates a patch for added list items', () => {
     const sourceEntry = sourceEntriesFixture.items[2] as unknown as Entry<any>
     const targetEntry = targetEntriesFixture.items[2] as unknown as Entry<any>
     const patch = createPatch({ targetEntry, sourceEntry })
     expect(patch).to.deep.equal([
+      {
+        op: 'replace',
+        path: '/fields/modules/en-US/4/sys/id',
+        value: '6gFiJvssqQ62CMYqECOu2M',
+      },
```

Comment:
By introducing the path check for `sys` only on root, we noticed that the old library didn't catch the nested id change within an array object. This is actually a known problem we already saw in other places with `jsondiffpatch`.